templates/beacon-node.service.j2

```diff
@@ -12,6 +12,12 @@ Restart=on-failure
 ExecStart={{ beacon_node_repo_path }}/build/nimbus_beacon_node \
     --network={{ beacon_node_network }} \
     --data-dir='{{ beacon_node_data_path }}/{{ beacon_node_data_folder }}' \
+{% if beacon_node_validators_dir != "" %}
+    --validators-dir='{{ beacon_node_validators_dir }}' \
+{% endif %}
+{% if beacon_node_secrets_dir != "" %}
```

Comment:
If variables are commented out they are undefined, and just an `if beacon_node_secrets_dir` check will work.

Reply:
Actually, both undefined variable and empty string are treated as false in a Jinja `if`.

Reply:
Scratch that. If `is defined` is not used Ansible throws `AnsibleUndefinedVariable`, so a proper check would be:
```j2
{% if beacon_node_secrets_dir is defined and beacon_node_secrets_dir != "" %}
```

Reply:
Truth be told, I don't really see why those should be optional. If we want to protect against a case in which those locations might change in the future in the code - however unlikely that might be - lets just set them to relative `validators` and `secrets` respectively by default and not worry about a case where they might change.